### ISSUE TITLE
ENH: Create project pages from github issues project proposals

### ISSUE THREAD

**Issue by MathewBiddle** [hackthackathon/hackathon-template-cookiecutter#8]
use github issue labels to create project pages. labels like accepted would work for this.

xref: https://github.com/BioEcoOcean/metadata-tracking-dev/issues/6

See 
https://github.com/BioEcoOcean/metadata-tracking-dev/blob/0980b1e26640a9539d303510581c75fbfae651bf/.github/workflows/gha_generate_pr.yml#L3-L6

https://github.com/BioEcoOcean/metadata-tracking-dev/blob/0980b1e26640a9539d303510581c75fbfae651bf/.github/workflows/gha_generate_pr.yml#L22-L24

Action would live at https://github.com/mgrover1/hackathon-template-cookiecutter/tree/main/%7B%7Bcookiecutter.hackathon_repo_name%7D%7D/.github/workflows so new repo's would have it available.

**Comment by MathewBiddle** [comment]
Something to start with:

```yaml
name: Create project page and Create Pull Request
  
on:
  issues:
    types: 
      - labeled


jobs:
  auto_new_resource_add_pr:
    if: contains(github.event.issue.labels.*.name, 'accepted')
    runs-on: ubuntu-latest
    steps:
      - name: Check Out Repository
        uses: actions/checkout@v4
        
      - name: Setup Micromamba ${{ matrix.python-version }}
        uses: mamba-org/setup-micromamba@v2
        with:
          environment-file: environment.yml
          init-shell: bash


      - name: Parsing issue, generate markdown
        shell: bash -l {0}
        run: python generate_pr_from_issue.py
      - name: Create Pull Request
        uses: peter-evans/create-pull-request@v7
        with:
          token: ${{ secrets.GITHUB_TOKEN }}
          commit-message: "Create new project page via GitHub Actions"
          committer: GitHub <noreply@github.com>
          author: ${{ github.actor }} <${{ github.actor }}@users.noreply.github.com>
          branch: new_project_#${{ github.event.issue.number }}
          title: "new project PR by GHA based on issue #${{ github.event.issue.number }}"
          body: |
            A project that is contributed by the user through issue submission. It has triggered a GitHub Action workflow that generate this PR. Refer to issue #${{ github.event.issue.number }}.
            Merging this PR will close #${{ github.event.issue.number }}.
          delete-branch: true
          labels: new project
```

**Comment by MathewBiddle** [comment]
myst allows for glob parameters in TOC description.

https://github.com/mgrover1/hackathon-template-cookiecutter/blob/60c97b2e6bb73dac746fcec74bd52fd59c11375e/%7B%7Bcookiecutter.hackathon_repo_name%7D%7D/myst.yml#L13-L18

Ideally, this action would read from the issue, generate a .md, stick it in a `projects/` folder. Then, myst should automatically pick it up and build into the website.

**Comment by MathewBiddle** [comment]
closed with #20
